### ISSUE TITLE
Does RedisCluster support password?

### ISSUE THREAD

**Issue by indusbull** [scredis/scredis#50]
I am using RedisCluster. It works fine when redis is not password protected. But when we make cluster protected and I add password in application.conf as mentioned in document. It doesn't seem to work.

```
scredis {
  redis {
    # Redis server address
    host = "hostname"
    # Redis server port
    #port = "7387"
    password = "redispassword"
    # List of hosts or host:port tuples, which serve as seed nodes for a Redis cluster connection.
    # If port is missing, the configured port is used as default.
    #cluster-nodes = [${scredis.redis.host}":"${scredis.redis.port}]
    cluster-nodes = ["hostname1:7385","hostname2:7386","hostname3:7387"]
    #cluster-nodes = ${?REDIS_CLUSTER_NODES}
  }
}
```

Then I use below code to initialized redis client -

`  RedisCluster redisClient = RedisCluster(providedActorSystem = providedActorSystem)`

I get an exception that password is required 

`The future returned an exception of type: scredis.exceptions.RedisErrorResponseException, with message: NOAUTH Authentication required..`

**Comment by indusbull** [comment]
Upon further debugging the code, I noticed that passwordOpt = None is being provided to AkkaNonBlockingConnection in [ClusterConnection.makeConnection](https://github.com/scredis/scredis/blob/4f35fbceac3a24aeb6e199ebd91d670adfe695b1/src/main/scala/scredis/io/ClusterConnection.scala#L111) 

Is there any reason that config password is not being used?


```
private def makeConnection(server: Server, systemOpt:Option[ActorSystem]): NonBlockingConnection = {
    val systemName = RedisConfigDefaults.IO.Akka.ActorSystemName
    val system = systemOpt.getOrElse(ActorSystem(UniqueNameGenerator.getUniqueName(systemName)))

    new AkkaNonBlockingConnection(
      system = system, host = server.host, port = server.port, passwordOpt = None,
      database = 0, nameOpt = None, decodersCount = 2,
      receiveTimeoutOpt, connectTimeout, maxWriteBatchSize, tcpSendBufferSizeHint,
      tcpReceiveBufferSizeHint, akkaListenerDispatcherPath, akkaIODispatcherPath,
      akkaDecoderDispatcherPath, failCommandOnConnecting
    ) {}
  }
```

**Comment by jastice** [comment]
If I recall correctly passwords weren't supported for clusters at the time this was implemented. Has this changed?

**Comment by indusbull** [comment]
We are able to set up redis cluster with password based on steps outlined [here](https://www.digitalocean.com/community/tutorials/how-to-configure-a-redis-cluster-on-ubuntu-14-04)

Just to test out quickly, I downloaded the scredis source code and updated that line of code to start passing passwordOpt. And it works! But not sure if there are cases where this might not be enough.

**Comment by kpbochenek** [comment]
As authorization for cluster connection was not supported at all this definitely improves current state of things, hopefully if will cover all cases we will see
Could you create a PR for actually using a password from configuration?

It can be more tricky when you have different passwords for different instances or you want to change password using `auth` message, but something to be checked later.

Also thanks for reporting this issue :+1:

**Comment by indusbull** [comment]
Sorry for late response. I am tied up with some high priority task. I will create PR  soon.

**Comment by kpbochenek** [comment]
I guess there will be no PR from indusbull, will try my best to work on that tomorrow

**Comment by indusbull** [comment]
My apologies for late response. I have created [pull request](https://github.com/scredis/scredis/pull/55). Please note that we have not tested this change in our production system yet.

**Comment by maciejbak85** [comment]
@kpbochenek as PR is openned can you after merge push to maven ? even some snapshot etc version, thanks!

**Comment by kpbochenek** [comment]
@maciejbak85 pls try `2.2.2+2-c1ad2e4e` (you need bintray resolver in sbt)
If all looks good for you I will do proper release.

https://bintray.com/scredis/maven/scredis/2.2.2+2-c1ad2e4e

**Comment by kpbochenek** [comment]
Fix released with version `2.2.3` and pushed to maven central.
Closing this issue, if you find something not working please open a new one.

**Comment by maciejbak85** [comment]
yes I`ve tried already :) Works fine ! thanks
